flow360/component/surface_mesh.py

```diff
@@ -194,18 +210,129 @@ def surface_mesh_file(self) -> str:
         """surface mesh file"""
         return self._surface_mesh_file
 
-    def _get_mesh_format(self) -> Union[None, SurfaceMeshFileFormat]:
-        mesh_format = None
-        if self.surface_mesh_file is not None:
-            mesh_format = SurfaceMeshFileFormat.detect(self.surface_mesh_file)
-        elif (
-            (self.geometry_file is not None or self.geometry_id is not None)
-            and Flags.beta_features()
-            and self.params is not None
-            and self.params.version == "v2"
+    # pylint: disable=protected-access
+    def _submit_from_geometry(self, progress_callback=None, force_submit: bool = False):
+        if not force_submit:
+            self.validator_api(self.params, solver_version=self.solver_version)
+
+        name = self.name
+        if name is None:
+            name = os.path.splitext(os.path.basename(self.geometry_file))[0]
+
+        mesh_format = SurfaceMeshFileFormat.UGRID
+        endianness = UGRIDEndianness.LITTLE
+        compression = CompressionFormat.NONE
+
+        stem = SURFACE_MESH_NAME_STEM_V1
+        if Flags.beta_features():
+            if self.params is not None and self.params.version == "v2":
+                stem = SURFACE_MESH_NAME_STEM_V2
+
+        if Flags.beta_features():
+            req = NewSurfaceMeshRequest(
+                name=name,
+                stem=stem,
+                tags=self.tags,
+                geometry_id=self.geometry_id,
+                config=self.params.flow360_json(),
+                mesh_format=mesh_format.value,
+                endianness=endianness.value,
+                compression=compression.value,
+                solver_version=self.solver_version,
+                version=self.params.version,
+            )
+        else:
+            req = NewSurfaceMeshRequest(
+                name=name,
+                stem=stem,
+                tags=self.tags,
+                geometry_id=self.geometry_id,
+                config=self.params.flow360_json(),
+                mesh_format=mesh_format.value,
+                endianness=endianness.value,
+                compression=compression.value,
+                solver_version=self.solver_version,
+            )
+        resp = RestApi(SurfaceMeshInterface.endpoint).post(req.dict())
+        info = SurfaceMeshMeta(**resp)
+        self._id = info.id
+        submitted_mesh = SurfaceMesh(self.id)
+
+        if self.geometry_file is not None:
+            _, ext = os.path.splitext(self.geometry_file)
+            remote_file_name = f"geometry{ext}"
+            file_name_to_upload = self.geometry_file
+            submitted_mesh._upload_file(
+                remote_file_name, file_name_to_upload, progress_callback=progress_callback
+            )
+            submitted_mesh._complete_upload(remote_file_name)
```

Comment:
why do we have this here? Upload geometry should be handled by Geometry class

Reply:
fixed